internal/storage/metricstore/elasticsearch/reader.go

```diff
@@ -22,15 +23,17 @@ const minStep = time.Millisecond
 
 // MetricsReader is a Elasticsearch metrics reader.
 type MetricsReader struct {
+	client es.Client
 	logger *zap.Logger
 	tracer trace.Tracer
 }
 
-func NewMetricsReader(_ config.Configuration, logger *zap.Logger, tracer trace.TracerProvider) (*MetricsReader, error) {
+func NewMetricsReader(_ config.Configuration, logger *zap.Logger, tracer trace.TracerProvider, client es.Client) *MetricsReader {
```

Comment:
* if config is not needed don't pass it
* client should be first param
* is there a decorator at a higher level that emits metrics about the calls to MetricReader? There has to be observability about those calls.